plugins/right-sizer/src/controller/controller.go

```diff
@@ -0,0 +1,271 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	fwControllerUtils "github.com/fairwindsops/controller-utils/pkg/controller"
+	"github.com/fairwindsops/insights-plugins/right-sizer/src/report"
+	"github.com/fairwindsops/insights-plugins/right-sizer/src/util"
+	"github.com/golang/glog"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+)
+
+const (
+	// informerSyncMinute defines how often the cache is synced from Kubernetes
+	informerSyncMinute = 2
+	// TerminationReasonOOMKilled is the reason of a ContainerStateTerminated that reflects an OOM kill
+	TerminationReasonOOMKilled = "OOMKilled"
+)
+
+// Controller is a controller that listens on Pod changes and create Kubernetes Events
+// when a container reports it was previously killed
+type Controller struct {
```

Comment:
there's a lot going on in this struct. I might suggest breaking it down into some sub-structs (e.g. you could group the k8s API stuff into a sub-struct).

Reply:
@rbren How's this? See [this commit](https://github.com/FairwindsOps/insights-plugins/commit/6f174f8839c5b46fd9b4d62a78cc7d35644d0e88) where I broke out the client, dynamicClient, and RESTMapper as those are already instantiated and provided at the same time.

This makes the `Controller` struct a bit nicer, but also use of the client/dynamicClient/RESTMapper resources a bit more verbose E.G. `c.kubeClientResources.Client`